src/plone/restapi/serializer/dxcontent.py

```diff
@@ -84,6 +86,24 @@ def __call__(self, version=None, include_items=True):
                 value = serializer()
                 result[json_compatible(name)] = value
 
+        # Fetch discussion registry
+        registry = queryUtility(IRegistry)
+        settings = registry.forInterface(IDiscussionSettings, check=False)
+
+        # Check if discussion is allowed globally, overwritting whatever that
+        # is set in the object
+        if not settings.globally_enabled:
+            result['allow_discussion'] = False
+        else:
+            if 'allow_discussion' not in result or \
+               result.get('allow_discussion', False) is None:
```

Comment:
`result.get('allow_discussion', None) is None` does the same thing as both lines above

Reply:
I've just improved the condition a bit and added an additional test for it.